src/sql/magic_cmd.py

```diff
@@ -40,33 +40,41 @@ def execute(self, line="", cell="", local_ns=None):
         """
         Command
         """
-        split = arg_split(line)
-        cmd_name, others = split[0].strip(), split[1:]
 
-        if cmd_name == "tables":
-            parser = CmdParser()
-
-            parser.add_argument(
-                "-s", "--schema", type=str, help="Schema name", required=False
-            )
-
-            args = parser.parse_args(others)
-
-            return inspect.get_table_names(schema=args.schema)
-        elif cmd_name == "columns":
-            parser = CmdParser()
-
-            parser.add_argument(
-                "-t", "--table", type=str, help="Table name", required=True
-            )
-            parser.add_argument(
-                "-s", "--schema", type=str, help="Schema name", required=False
-            )
-
-            args = parser.parse_args(others)
-            return inspect.get_columns(name=args.table, schema=args.schema)
-        else:
+        if line == "":
             raise UsageError(
```

Comment:
I think we can extract this logic to another function, e.g : `_validate_inputs`. there we can check the inputs (empty or not table/column) and raise the relevant error. this way we can control the error message in one place and re-use some of the code. 

After that, we can safely continue with our code (since we know our inputs are valid)

Reply:
Have added the request function in the code. Have also updated the test_magic_cmd.py. 

Please review the code and let me know.

Thanks and regards,
Raghav